src/ImpactEvaluator.sol

```diff
@@ -89,40 +88,15 @@ contract ImpactEvaluator is AccessControl {
         Round storage round = openRounds[roundIndex];
         require(round.exists, "Open round does not exist");
 
-        if (moreScoresExpected) {
-            for (uint i = 0; i < addresses.length; i++) {
-                round.addresses.push(addresses[i]);
-                round.scores.push(scores[i]);
-            }
-        } else {
-            (
-                address payable[] memory mergedAddresses,
-                uint64[] memory mergedScores
-            ) = mergeScores(round, addresses, scores);
-            reward(mergedAddresses, mergedScores);
-            cleanUpRound(round, roundIndex);
+        for (uint i = 0; i < addresses.length; i++) {
+            round.addresses.push(addresses[i]);
+            round.scores.push(scores[i]);
         }
-    }
 
-    function mergeScores(
-        Round storage round,
-        address payable[] memory addresses,
-        uint64[] memory scores
-    ) private view returns (address payable[] memory, uint64[] memory) {
-        uint mergedLength = round.addresses.length + addresses.length;
-        address payable[] memory mergedAddresses = new address payable[](
-            mergedLength
-        );
-        uint64[] memory mergedScores = new uint64[](mergedLength);
-        for (uint i = 0; i < round.addresses.length; i++) {
-            mergedAddresses[i] = round.addresses[i];
-            mergedScores[i] = round.scores[i];
-        }
-        for (uint i = 0; i < addresses.length; i++) {
-            mergedAddresses[round.addresses.length + i] = addresses[i];
-            mergedScores[round.addresses.length + i] = scores[i];
+        if (allScoresSubmitted(round.scores)) {
+            reward(round.addresses, round.scores);
```

Comment:
The last call of `setScores` for the given round will calculate the sum of scores twice: first time inside `setScores/allScoresSubmitted`, and second time inside  `reward/validateScores`.

Is it worth optimising this part to reduce the gas cost?

1. Depending on the storage cost, the contract can remember the sum of all scores submitted so far. The function `setScores` can then update this sum by iterating over new scores only. 

   Benefit: we spread the cost of iterating over all scores over several `setScore` calls. This matters if different parties can submit scores, because everybody pays for iterating over their scores. If somebody submits a large list of participants, they pay the gas fee, not the actor submitting the last batch.

2. Alternatively, we can move everything related to `sum(scores)` into the `setScores` function. (In particular, move the validation step.) That way we can compute the sum only once and refer to it multiple times.

3. I suppose we can also pass the sum around in arguments to `reward` and `validateScores`, but I find that a bit ugly.